stdlib/public/runtime/ImageInspectionELF.cpp

```diff
@@ -72,9 +72,14 @@ static SectionInfo getSectionInfo(const char *imageName,
   SectionInfo sectionInfo = { 0, nullptr };
   void *handle = dlopen(imageName, RTLD_LAZY | RTLD_NOLOAD);
   if (!handle) {
-#ifdef __ANDROID__
+#if defined(__ANDROID__) || defined(__FreeBSD__)
+#if defined(__ANDROID__)
```

Comment:
Not extremely happy about this macro dance but it seems consistent with what's there (and probably OK).

Reply:
Why is this necessary on FreeBSD? It is necessary on Android because the OS hides the contents of `/system/lib` from `dlopen()`.

Reply:
I'm not sure about the reasons but this part is called and fails with `/libexec/ld-elf.so.1` and then with the main executable path. As for the latter, from https://github.com/apple/swift/pull/10836#issuecomment-317014783 I assumed that `dl_iterate_phdr` has the same behavior on FreeBSD.